src/test/scala/GameServiceTest.scala

```diff
@@ -0,0 +1,96 @@
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funspec.AnyFunSpec
+import org.scalamock.scalatest.MockFactory
+import service.GameServiceImpl
+import service.GameService
+import persistance.GameRepository
+import command.Move
+import command.Move._
+import model.GameResult._
+import persistance.GameStatusEntity
+import service.CpuMoveStrategy
+
+class GameServiceTest extends AnyFunSpec with MockFactory with BeforeAndAfterEach {
+
+  var mockgameRepository: GameRepository = _
+  var mockCpuMoveStrategy: CpuMoveStrategy = _
+  var gameService: GameServiceImpl = _
+
+  override def beforeEach() {
+    mockgameRepository = mock[GameRepository]
+    mockCpuMoveStrategy = mock[CpuMoveStrategy]
+    gameService = new GameServiceImpl(mockgameRepository, mockCpuMoveStrategy)
+  }
+
+  describe("Test play") {
+
+    it("user lose") {
+
+      (mockCpuMoveStrategy.provideCPUMove _).expects().returns(Paper)
+      (mockgameRepository.insertGameResult _).expects(Rock, Paper, Lose).once()
+
+      val actualResult = gameService.play(Rock)
```

Comment:
this is the part I really don't like, but it's not 100% clear how to solve it. Theoretically, you'd want to test the game logic in isolation ("given this CPU move and this user move, the expected result is ..."), but even for this simple case you need to mock the `CpuMoveStrategy` (quite acceptable) and, above all, the repository (baaaad 😱)

I would prefer to have a pure `GameLogic` (tested completely in isolation) and some other class that handles the orchestration (generate the CPU move -> calculate the result -> store the match result), so that you can have unit tests on GameLogic (no mocking) and integration tests on the orchestrator (just check that the right methods are called, but without checking the actual game result)